### PR TITLE
Add a "Now Playing" segment

### PR DESCRIPTION
**Pull request by elken**
Implement a segment for showing the current output of playerctl.

Uses the same logic as `github` to run the commands async and truncates if it's too long.

I welcome any criticism! :)

----

#

### CODE REVIEW

File: doom-modeline-segments.el [hunk 1]
```diff
@@ -2090,6 +2090,69 @@ mouse-3: Fetch notifications"
...
+                             :face 'doom-modeline-now-playing-other-icons
+                             :v-adjust -0.0575))
+      (doom-modeline-spc)
+      (if (equal status "playing")
+           (doom-modeline-icon 'faicon "play" "" ">"
+                               :v-adjust -0.0575)
+           (doom-modeline-icon 'faicon "pause" "" "||"
+                               :v-adjust -0.0575))
+      (doom-modeline-spc)
+      (propertize (truncate-string-to-width text 50 nil nil "...") 'face 'doom-modeline-now-playing-text)))))
```
Comment (seagle0128) [inline doom-modeline-segments.el:2154]: it's better make 50 to an option.

File: doom-modeline-segments.el [hunk 2]
```diff
@@ -2090,6 +2090,69 @@ mouse-3: Fetch notifications"
...
+(defun doom-modeline--now-playing-update ()
+  "Update the tokens for now-playing."
+  (when (and doom-modeline-now-playing
+             (require 'async nil t))
+    (async-start
+     `(lambda ()
+        (require 'subr-x)
+        ,(async-inject-variables
+          "\\`\\(load-path\\|auth-sources\\|doom-modeline-now-playing-format\\|doom-modeline-now-playing-ignored-players\\)\\'")
+        (string-trim (shell-command-to-string (format "playerctl metadata --ignore-player=%s --format '{{playerName}}|{{lc(status)}}|%s'" (mapconcat 'identity doom-modeline-now-playing-ignored-players ",") doom-modeline-now-playing-format))))
```
Comment (seagle0128) [inline doom-modeline-segments.el:2107]: Does it support all platforms?

File: doom-modeline-core.el [hunk 3]
```diff
@@ -427,6 +427,28 @@ Non-nil to display in the mode-line."
   :type 'boolean
   :group 'doom-modeline)
 
+(defcustom doom-modeline-now-playing t
```
Comment (seagle0128) [inline doom-modeline-core.el:430]: I prefer the default value is nil.

**Review by seagle0128 — CHANGES_REQUESTED** [review]

File: doom-modeline-segments.el (hunk 1 above)
Comment (elken) [reply doom-modeline-segments.el:2154]: Yeah I realised this morning that it should be configurable, I'll move it out when I get home

File: doom-modeline-segments.el (hunk 2 above)
Comment (elken) [reply doom-modeline-segments.el:2107]: playerctl should work on any *NIX platform, I thought about making an adapter to making it more xplatform. If you would prefer me to do so I will :)

File: doom-modeline-core.el (hunk 3 above)
Comment (elken) [reply doom-modeline-core.el:430]: I will also add this tonight

File: doom-modeline-segments.el (hunk 2 above)
Comment (seagle0128) [reply doom-modeline-segments.el:2107]: yep, I'd like it works on all platforms, Windows, Linux, macOS.

Comment (elken) [reply doom-modeline-segments.el:2107]: Roger that, I'll look into writing an adapter then

**Comment by elken** [comment]
I'll take a crack at writing some adapter middleware over the next few days, thanks for the feedback! :)

**Comment by elken** [comment]
Finally gotten around to looking at this, this could be made to work with mac and linux both with relative ease, however the only feasible way for it to work with Windows is via the web API which would require end-user configuration.

This is the most portable way but means it doesn't work ootb, if this is satisfactory I will work on adding this, otherwise I guess you should close this ^^

**Comment by seagle0128** [comment]
OK. I am closing it. Please feel free to reopen it if you have a new method.

**Pull request closed without merging**